src/assertions/utils.ts

```diff
@@ -9,11 +9,26 @@ const clone = Clone();
 
 export function getFinalTest(test: TestCase, assertion: Assertion) {
   // Deep copy
-  const ret = clone(test);
+  const ret = clone({
+    ...test,
+    ...(test.options &&
+      test.options.provider && {
+        options: {
+          ...test.options,
+          provider: undefined,
```

Comment:
are you sure this isn't going to bite us one day?  I would expect these fields to be set.  Can we coerce them to json-safe values for example?

Reply:
Hopefully the test coverage is sufficient. Check lines 28-32 we write to the fields again with the original values